lib/endpoints/class-wp-json-comments-controller.php

```diff
@@ -47,6 +47,10 @@ public function get_item( $request ) {
 			return new WP_Error( 'json_comment_invalid_id', __( 'Invalid comment ID.' ), array( 'status' => 404 ) );
 		}
 
+		if ( ! empty( $request['context'] ) && 'edit' === $request['context'] && ! current_user_can( 'moderate_comments' ) ) {
```

Comment:
We should move this check to the permissions check callback instead me things now